From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vincent Strubel <clipos@ssi.gouv.fr>
Date: Wed, 11 Jun 2014 18:19:48 +0200
Subject: [PATCH 07/32] clip: vserver: Allow WATCH context to see every
 /proc/<pid>

Patch: 1506_vserver_allow_watch_context_proc_pid.patch
---
 fs/proc/base.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fs/proc/base.c b/fs/proc/base.c
index cd8d28cf0179e..e45fdc006fb00 100644
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -753,6 +753,9 @@ static bool has_pid_permissions(struct pid_namespace *pid,
 	}
 	rcu_read_unlock();
 
+	if (vx_check(0, VS_WATCH_P))
+		return true;
+
 	if (!pid->hide_pid)
 		return ptrace_may_access(task, PTRACE_MODE_READ_FSCREDS | PTRACE_MODE_NOAUDIT);
 #endif
-- 
2.17.0

